### DISCUSSION TITLE
AWS services used by Magma orc8r and AGW

### DISCUSSION THREAD

**Discussion by jimhuangh** [magma/magma#7412] [General]
awsAccessKey and awsSecretKey specified in orc8r's secrets.yaml file enable orc8r to interact with AWS services. What're the exact AWS services which orc8r uses via awsAccessKey and awsSecretKey? 

AWS IAM policy needs to be specified to define the scope of orc8r's access to AWS services. Can the Magma team please provide the names of AWS services used by orc8r with awsAccessKey and awsSecretKey? 

Thanks,
Jim 
PS. Magma Team, you may look up standardized AWS service names through your AWS account at IAM Create Policy.

**Comment by arunuke** [comment]
hello @jimhuangh - As for a list of AWS services, I would say Elasticsearch, EKS, EC2 (asgs, elbs), RDS, EFS, secrets, cloudwatch and VPC (internet and nat gateways, subnets, security groups). Adding @karthiksubraveti and @hcgatewood for any additional inputs.

**Reply by jimhuangh** [reply]
Magma Team,

AWS Lab requires the solution owner (Magma Team) to provide a Policy file in JSON that describes assess permissions for the AWS services the solution (Magma) uses with awsAccessKey and awsSecretKey. 

You may use AWS IAM Policy Visual Editor to assist policy creation and modification: https://aws.amazon.com/blogs/security/use-the-new-visual-editor-to-create-and-modify-your-aws-iam-policies/.

Thanks,
Jim

**Reply by arunuke** [reply]
Hello Jim - the hardening of the deployment environment is an ongoing discussion that ie being tracked and is a task typically taken up by the System Integrator (SI) as part of the production deployment. The AWS deploy in the lab is a POC from what I remember, but if it's an absolute requirement for this environment hosted and managed by AWS, you could start by enabling access for the services I listed above.

**Reply by AdamIsrael** [reply]
For the AGW, which is the component we're trying to deploy to the lab, it looks like the only policies that are needed are:
- AmazonEC2FullAccess
- AWSCloudFormationFullAccess

With just those permissions set, I've been able to deploy the AGW in a non-lab setting.

I'm working to validate the policies required for the Orchestrator, but since that component will run from outside the lab, the above should unblock us.

Thanks, Arun and Jim.

**Reply by arunuke** [reply]
Great point and thanks @AdamIsrael . The AGWs themselves require only EC2 services to run the AMI, VPC resources for the networking (which are presumably covered under the EC2 service) and Cloudformation to orchestrate them end to end.

**Comment by jimhuangh** [comment]
@AdamIsrael<https://github.com/AdamIsrael> Since we run orc8r in the Outposts account, Policy for orc8r’s access to AWS services is also required.

From: Arun Thulasi ***@***.***>
Reply-To: magma/magma ***@***.***>
Date: Wednesday, June 9, 2021 at 1:45 PM
To: magma/magma ***@***.***>
Cc: "Huang, Jim" ***@***.***>, Mention ***@***.***>
Subject: Re: [magma/magma] AWS services used by Magma orc8r and AGW (#7412)


Great point and thanks @AdamIsrael<https://github.com/AdamIsrael> . The AGWs themselves require only EC2 services to run the AMI, VPC resources for the networking (which are presumably covered under the EC2 service) and Cloudformation to orchestrate them end to end.

**Reply by jimhuangh** [reply]
@arunuke Could Magma Team please clarify if orc8r deployment and operation use any AWS services through awsAccessKey and awsSecretKey?

**Reply by arunuke** [reply]
orchestrator services use AWS credentials that are either stored in secrets.yaml (if using Cloudstrapper) or in the user's AWS profile (~/.aws/credentials).

**Comment by jimhuangh** [comment]
@arunuke For the AGW, please confirm this IAM policy is correct with the Magma release (v1.5.0).

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "EC2FullAccess",
            "Action": "ec2: *",
            "Resource":  "*",
            "Effect": "Allow",
            "Condition": {
                "StringEquals": {
                    "ec2:Region": "us-west-2"
                }
            }
        },
        {
            "Sid": "FloudFormationFullAccess",
            "Action": "cloudformation:*",
            "Resource": "*",
            "Effect": "Allow",
            "Condition": {
                "StringEquals": {
                    "ec2:Region": "us-west-2"
                }
            }
        }
    ]
}
